docs/src/cookbooks/cookbook.md

```diff
@@ -66,6 +66,28 @@ class Foo extends RawModule {
 }
 ```
 
+### How can I tieoff a Bundle/Vec to 0?
+
+You can use `asTypeOf` as above, but also use `chiselTypeOf` rather than having to
+figure out the type of the LHS:
+
+```scala mdoc:silent:reset
+import chisel3._
+
+class MyBundle extends Bundle {
+  val foo = UInt(4.W)
+  val bar = Vec(4, UInt(1.W))
+}
+
+class Foo(gen: () => Data) extends RawModule {
+  val bundle = IO(Output(gen))
+  bundle := 0.U.asTypeOf(chiselTypeOf(gen))
```

Comment:
```suggestion
  bundle := 0.U.asTypeOf(gen)
```

Reply:
if this is what i should write this whole example is pointless, the point was to show needing to call `chiselTypeOf`

Reply:
Haha, right, sorry, yeah that makes sense.

Reply:
well, if this works without the `chiselTypeOf` then that's probably not a great cookbook... maybe what I mean is:

```suggestion
  bundle := 0.U.asTypeOf(chiselTypeOf(bundle))
```

Reply:
Yes I think that's better

Reply:
I think what I'll do is show both with some comments -- bundle is a hardware type, while gen is already a chisel type